### PR TITLE
DEV: Refactor `ThemeSettingsObjectValidator#validate`

### PR DESCRIPTION
**Pull request by tgxworld**
What does this change do?

1. Reduce an additional loop through all the properties
2. Extract the validation of child objects into a dedicate method

### CODE REVIEW

File: lib/theme_settings_object_validator.rb [hunk 1]
```diff
@@ -94,15 +87,30 @@ def validate
 
   private
 
-  def validate_properties
-    @properties.each do |property_name, property_attributes|
```
Comment (tgxworld) [inline lib/theme_settings_object_validator.rb:98]: This is the extra loop we are eliminating

File: lib/theme_settings_object_validator.rb [hunk 2]
```diff
@@ -94,15 +87,30 @@ def validate
...
+          valid_category_ids:,
+          json_pointer_prefix: "#{@json_pointer_prefix}#{property_name}/#{index}/",
+          errors: @errors,
+        )
+        .validate
     end
   end
 
+  def validate_property(property_name, property_attributes)
+    return if property_attributes[:type] == "objects"
```
Comment (OsamaSayegh) [inline lib/theme_settings_object_validator.rb:108]: I think we can delete this line since this method is only called when the type isn't `objects`.

**Review by OsamaSayegh — APPROVED** [review]

File: lib/theme_settings_object_validator.rb (hunk 2 above)
Comment (tgxworld) [reply lib/theme_settings_object_validator.rb:108]: O good catch 👍

**Comment by tgxworld** [comment]
Thank you @OsamaSayegh 👍

**Pull request merged**